src/vs/workbench/contrib/notebook/browser/notebookOptions.ts

```diff
@@ -145,7 +145,7 @@ export class NotebookOptions extends Disposable {
 		const cellToolbarInteraction = overrides?.cellToolbarInteraction ?? this.configurationService.getValue<string>(NotebookSetting.cellToolbarVisibility);
 		const compactView = this.configurationService.getValue<boolean | undefined>(NotebookSetting.compactView) ?? true;
 		const focusIndicator = this._computeFocusIndicatorOption();
-		const insertToolbarPosition = this._computeInsertToolbarPositionOption();
+		const insertToolbarBetweenCells = overrides?.insertToolbarBetweenCells ?? this._computeInsertToolbarBetweenCellsOption();
```

Comment:
should we check if the notebook document is readonly or not?

Reply:
Yep, it is a bigger change now but it looks great. I used `Show Contents` on local history timeline/git timeline to see a readonly jupyter notebook.

Reply:
`insertToolbarBetweenCells` should be set to false here if the notebook is readonly. That would eliminate the need to pass around the readonly parameter in all those other methods.

Reply:
That information comes from the model, `NotebookEditorWidget` doesn't have the model in its constructor, only later when `NotebookEditorWidget.setModel` is called (which can be done multiple times)

Here's the call stack when using Show Contents on a git revision (`creationOptions` is ```{
  menuIds: {...
  },
  cellEditorContributions: [
...]
}```)

![image](https://github.com/microsoft/vscode/assets/1485998/67cac52f-c0b9-4032-bfb8-b367d429054b)

Then setModel-
![image](https://github.com/microsoft/vscode/assets/1485998/0c896ab1-fdde-494b-9793-75ade376d945)

Then setOptions with isReadOnly set-

![image](https://github.com/microsoft/vscode/assets/1485998/d9eba377-f25f-4f4b-bd09-8534528b05d7)